README.md
# responsive

This is a resposive extension which allow you to spefic screen and screensize in your flutter application. You do not need to remember size width.

## Example

```
import 'package:flutter/material.dart';
import 'package:responsive_context/responsive_context.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      initialRoute: '/',
      routes: {
        '/': (_) => DatePickerPage(),
      },
    );
  }
}

class DatePickerPage extends StatefulWidget {
  @override
  _DatePickerPageState createState() => _DatePickerPageState();
}

class _DatePickerPageState extends State<DatePickerPage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("Responsive Context"),
      ),
      body: SingleChildScrollView(
        child: Container(
          child: Column(
            mainAxisSize: MainAxisSize.max,
            mainAxisAlignment: MainAxisAlignment.start,
            children: [
              Container(
                width: 600,
                // height: 600,
                // width: MediaQuery.of(context).size.width,
                child: LayoutBuilder(
                  builder: (ctx, constrain) {
                    // return Text("${!ctx.isExtraLarge? "HIDE" : "SHOW"}");
                    return GridView(
                        shrinkWrap: true,
                        padding: EdgeInsets.all(15),
                        gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
                            crossAxisCount: constrain.isExtraSmall
                                ? 1
                                : constrain.isSmall
                                    ? 2
                                    : constrain.isMedium
                                        ? 3
                                        : constrain.isLarge ? 4 : 5,
                            crossAxisSpacing: 15,
                            mainAxisSpacing: 15),
                        children: List.filled(
                            5,
                            Container(
                              color: Colors.blue,
                              alignment: Alignment.center,
                              child: Text(
                                "${constrain.screenSize} based on constrain",
                                textAlign: TextAlign.center,
                              ),
                            )).toList());
                  },
                ),
              ),
              Container(
                width: 600,
                // height: 600,
                // width: MediaQuery.of(context).size.width,
                child: GridView(
                    shrinkWrap: true,
                    padding: EdgeInsets.all(15),
                    gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
                        crossAxisCount: context.isExtraSmall
                            ? 1
                            : context.isSmall
                                ? 2
                                : context.isMedium
                                    ? 3
                                    : context.isLarge ? 4 : 5,
                        crossAxisSpacing: 15,
                        mainAxisSpacing: 15),
                    children: List.filled(
                        5,
                        Container(
                          color: Colors.red,
                          alignment: Alignment.center,
                          child: Text(
                            "${context.screenSize} based on context",
                            textAlign: TextAlign.center,
                          ),
                        )).toList()),
              ),
            ],
          ),
        ),
      ),
    );
  }
}

```

This project is a starting point for a Dart
[package](https://flutter.dev/developing-packages/),
a library module containing code that can be shared easily across
multiple Flutter or Dart projects.

For help getting started with Flutter, view our 
[online documentation](https://flutter.dev/docs), which offers tutorials, 
samples, guidance on mobile development, and a full API reference.


CHANGELOG.md
# [0.2.0]

* nullable support
* effective dart

# [0.1.2]

* fix uri

## [0.1.1]

* edit xs and add maxWidth to context

## [0.1.0] - TODO: Add release date.

* TODO: Describe initial release.


test/responsive_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:responsive_context/responsive_context.dart';

void main() {
  test('run without test', () {
    
  });
}


lib/ExtensionBuildConext.dart
import 'package:flutter/widgets.dart';

import 'ScreenSize.dart';

extension ExtensionBuildConext on BuildContext {
  bool get isLandscape =>
      MediaQuery.of(this).orientation == Orientation.landscape;
  bool get isPortrait =>
      MediaQuery.of(this).orientation == Orientation.portrait;
  bool get isExtraSmall => MediaQuery.of(this).size.width <= 600;
  bool get isSmall =>
      MediaQuery.of(this).size.width > 600 &&
      MediaQuery.of(this).size.width <= 768;
  bool get isMedium =>
      MediaQuery.of(this).size.width > 768 &&
      MediaQuery.of(this).size.width <= 992;
  bool get isLarge =>
      MediaQuery.of(this).size.width > 992 &&
      MediaQuery.of(this).size.width <= 1200;
  bool get isExtraLarge => MediaQuery.of(this).size.width > 1200;
  double get maxWidth => MediaQuery.of(this).size.width;
  String get screen {
    if (isExtraSmall) return "ExtraSmall";
    if (isSmall) return "Small";
    if (isMedium) return "Medium";
    if (isLarge) return "Large";
    if (isExtraLarge) return "ExtraLarge";
    return "Unknown";
  }

  ScreenSize get screenSize {
    if (isExtraSmall) return ScreenSize.xs;
    if (isSmall) return ScreenSize.sm;
    if (isMedium) return ScreenSize.md;
    if (isLarge) return ScreenSize.lg;
    if (isExtraLarge) return ScreenSize.xl;
    return ScreenSize.unknown;
  }
}


lib/ExtensionBoxConstrain.dart
import 'package:flutter/widgets.dart';

import 'ScreenSize.dart';

extension ExtensionBoxConstrain on BoxConstraints {
  bool get isExtraSmall => maxWidth <= 600;
  bool get isSmall => maxWidth > 600 && maxWidth <= 768;
  bool get isMedium => maxWidth > 768 && maxWidth <= 992;
  bool get isLarge => maxWidth > 992 && maxWidth <= 1200;
  bool get isExtraLarge => maxWidth > 1200;

  String get screen {
    if (isExtraSmall) return "ExtraSmall";
    if (isSmall) return "Small";
    if (isMedium) return "Medium";
    if (isLarge) return "Large";
    if (isExtraLarge) return "ExtraLarge";
    return "Unknown";
  }

  ScreenSize get screenSize {
    if (isExtraSmall) return ScreenSize.xs;
    if (isSmall) return ScreenSize.sm;
    if (isMedium) return ScreenSize.md;
    if (isLarge) return ScreenSize.lg;
    if (isExtraLarge) return ScreenSize.xl;
    return ScreenSize.unknown;
  }
}


example/main.dart
import 'package:flutter/material.dart';
import 'package:responsive_context/responsive_context.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      initialRoute: '/',
      routes: {
        '/': (_) => DatePickerPage(),
      },
    );
  }
}

class DatePickerPage extends StatefulWidget {
  @override
  _DatePickerPageState createState() => _DatePickerPageState();
}

class _DatePickerPageState extends State<DatePickerPage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("Responsive Context"),
      ),
      body: SingleChildScrollView(
        child: Container(
          child: Column(
            mainAxisSize: MainAxisSize.max,
            mainAxisAlignment: MainAxisAlignment.start,
            children: [
              Container(
                width: 600,
                // height: 600,
                // width: MediaQuery.of(context).size.width,
                child: LayoutBuilder(
                  builder: (ctx, constrain) {
                    // return Text("${!ctx.isExtraLarge? "HIDE" : "SHOW"}");
                    return GridView(
                        shrinkWrap: true,
                        padding: EdgeInsets.all(15),
                        gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
                            crossAxisCount: constrain.isExtraSmall
                                ? 1
                                : constrain.isSmall
                                    ? 2
                                    : constrain.isMedium
                                        ? 3
                                        : constrain.isLarge ? 4 : 5,
                            crossAxisSpacing: 15,
                            mainAxisSpacing: 15),
                        children: List.filled(
                            5,
                            Container(
                              color: Colors.blue,
                              alignment: Alignment.center,
                              child: Text(
                                "${constrain.screenSize} based on constrain",
                                textAlign: TextAlign.center,
                              ),
                            )).toList());
                  },
                ),
              ),
              Container(
                width: 600,
                // height: 600,
                // width: MediaQuery.of(context).size.width,
                child: GridView(
                    shrinkWrap: true,
                    padding: EdgeInsets.all(15),
                    gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
                        crossAxisCount: context.isExtraSmall
                            ? 1
                            : context.isSmall
                                ? 2
                                : context.isMedium
                                    ? 3
                                    : context.isLarge ? 4 : 5,
                        crossAxisSpacing: 15,
                        mainAxisSpacing: 15),
                    children: List.filled(
                        5,
                        Container(
                          color: Colors.red,
                          alignment: Alignment.center,
                          child: Text(
                            "${context.screenSize} based on context",
                            textAlign: TextAlign.center,
                          ),
                        )).toList()),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
